### ISSUE TITLE
Cross-Calorimeter Topocluster Algorithm

### ISSUE THREAD

**Issue by ruse-traveler** [eic/EICrecon#1561]
**Is your feature request related to a problem? Please describe.**

Currently we have two algorithms used for forming topoclusters:

- `ImagingTopoCluster`, which combines hits based on proximity across multiple layers of a calorimeter; and
- `EnergyPositionClusterMerger`, which combines the positions of one set of clusters with the energies of another.

However, we don't have an algorithm which generically takes two collections (or possibly more) of clusters and merges them into topoclusters. Such functionality would be useful when one needs to combine clusters from multiple calorimeters, such as in Particle Flow or when considering observables like E_{T}^{miss}.

**Describe the solution you'd like**

This could be implemented as an algorithm which takes as input two collections of protoclusters and outputs a single collection of protoclusters (the topoclusters). The initial implementation could match clusters based simply on proximity, but could be refined later to take advantage of the segmentation of some calorimeters.

This would mean that the relevant parameters for the algorithm would be
- The weights associated with each input collection of clusters,
- And some sort of tolerance for matching.

**Describe alternatives you've considered**

There are a couple of alternative approaches one could imagine, for example:
- This could be something that is left up to analyzers after the fact, or
- This functionality could be reserved for another type of algorithm downstream (e.g. a more explicitly "particle flow" algorithm).

**Comment by wdconinc** [comment]
The work by @AkshayaVijay on simultaneous hybrid clustering of BIC AstroPix and ScFi hits may be relevant here too.

**Comment by ruse-traveler** [comment]
> The work by [@AkshayaVijay](https://github.com/AkshayaVijay) on simultaneous hybrid clustering of BIC AstroPix and ScFi hits may be relevant here too.

Ah! #2268 right? To be honest, this PR slipped past my radar, but it's absolutely relevant to this topic! We should discuss more in the Reco WG or Calo CCWG meeting!

**Comment by wdconinc** [comment]
> > The work by [@AkshayaVijay](https://github.com/AkshayaVijay) on simultaneous hybrid clustering of BIC AstroPix and ScFi hits may be relevant here too.
> 
> Ah! [#2268](https://github.com/eic/EICrecon/pull/2268) right? To be honest, this PR slipped past my radar, but it's absolutely relevant to this topic! We should discuss more in the Reco WG or Calo CCWG meeting!

Yes, that's the PR (but I think there's some work that might not have made it in there, and there is still the unsettled discussion on how to pass this information in a scalable way in configuration. Maybe once the BIC test beam activity settles down, @AkshayaVijay can give a talk at a reconstruction meeting on this.

**Comment by ruse-traveler** [comment]
> Yes, that's the PR (but I think there's some work that might not have made it in there, and there is still the unsettled discussion on how to pass this information in a scalable way in configuration. Maybe once the BIC test beam activity settles down, [@AkshayaVijay](https://github.com/AkshayaVijay) can give a talk at a reconstruction meeting on this.

Definitely!
